samcli/commands/local/cli_common/invoke_context.py

```diff
@@ -154,6 +157,8 @@ def __init__(
             Optional. Docker extra hosts support from --add-host parameters
         invoke_images dict
             Optional. A dictionary that defines the custom invoke image URI of each function
+        ctx dict
+            Optional. Command invoke context used to invoke commands before invoking `sam local` logic
```

Comment:
If we are only using to grab the `debug` flag, maybe we can pass that field instead of the whole `Context` object?